src/protocol/commands.rs

```diff
@@ -177,6 +177,7 @@ commands! {
         "qsThreadInfo" => _qsThreadInfo::qsThreadInfo,
         "qSupported" => _qSupported::qSupported<'a>,
         "qXfer:features:read" => _qXfer_features_read::qXferFeaturesRead,
+        "qXfer:memory-map:read" => _qXfer_memory_map::qXferMemoryMapRead,
```

Comment:
This feature should be in a separate command group, as it isn't something that is supported by all targets.

You can take a peek at some past PRs (e.g: #48) for guidance on how to create a new command + handler group, but it ought to be pretty straightforward 😄